sample_size/sample_size_calculator.py

```diff
@@ -22,13 +23,15 @@ class SampleSizeCalculator:
 
     """
 
-    def __init__(self, alpha: float = DEFAULT_ALPHA, power: float = DEFAULT_POWER):
+    def __init__(self, alpha: float = DEFAULT_ALPHA, variants: int = DEFAULT_VARIANTS, power: float = DEFAULT_POWER):
         self.alpha = alpha
         self.power = power
+        self.variants = variants
         # Consider having a self.metrics to hold all metric types
         self.boolean_metrics: List[BooleanMetric] = []
         self.numeric_metrics: List[NumericMetric] = []
         self.ratio_metrics: List[RatioMetric] = []
+        self.rep: int = 100
 
     def register_bool_metric(self, probability: float, mde: float) -> None:
```

Comment:
Can we have a main `register_metrics(self, metrics: list[Dict[str, Any]])` in `sample_size_calculator` so that hivemind api can just pass in a list of metrics all together. `register_metrics` should take a list of metrics, the metrics should be dictionaries.

```
metrics = [
        {
            "metric_type": "boolean",
            "metric_metadata": {
                "probability": 0.05,
                "mde": 0.02,
            },
        },
    ],
```